osu.Game.Rulesets.Taiko/UI/DrumTouchInputArea.cs

```diff
@@ -65,27 +73,51 @@ private void load(TaikoInputManager taikoInputManager, OsuColour colours)
                             RelativeSizeAxes = Axes.Both,
                             Children = new Drawable[]
                             {
-                                leftRim = new QuarterCircle(TaikoAction.LeftRim, colours.Blue)
+                                leftRim = new QuarterCircle(touchControlsType.Value == TaikoTouchControlType.Default ? TaikoAction.LeftRim :
+                                                            touchControlsType.Value == TaikoTouchControlType.DDKK ? TaikoAction.LeftCentre :
+                                                          /*touchControlsType.Value == TaikoTouchControlType.KKDD*/ TaikoAction.LeftRim,
+
+                                                            touchControlsType.Value == TaikoTouchControlType.Default ? colours.Blue :
+                                                            touchControlsType.Value == TaikoTouchControlType.DDKK ? colours.Pink :
+                                                          /*touchControlsType.Value == TaikoTouchControlType.KKDD*/ colours.Blue)
```

Comment:
Avoid using multiple ternary expressions when possible. It's nigh impossible to mentally parse what's happening here.

I would instead move specifications inside `QuarterCircle` itself and branch drawable definitions based on the type, e.g.:
```csharp
switch (type)
{
    case TaikoTouchControlType.KDDK:
        mainContent.Children = new[]
        {
             firstCircle = new QuarterCircle(TaikoAction.LeftRim),
             secondCircle = new QuarterCircle(TaikoAction.LeftCentre),
             thirdCircle = new QuarterCircle(TaikoAction.RightCentre),
             fourthCircle = new QuarterCircle(TaikoAction.RightRim),
        };

        break;

    ...
}

private class QuarterCircle
{
    public QuarterCircle(TaikoAction action, Color4 colour)
    {
        Anchor = Anchor.BottomCentre;
        Origin = Anchor.BottomRight;

        if (action == TaikoAction.LeftCentre || action == TaikoAction.RightCentre)
            Scale = new Vector2(centre_region);
    }
}
```